### ISSUE TITLE
Cross-Platform Integration using `xtask-wasm`

### ISSUE THREAD

**Issue by antimora** [tracel-ai/burn#851]
**Description:**

As we aim to make our project more cross-platform compatible, leveraging [`xtask-wasm`](https://github.com/rustminded/xtask-wasm) appears to be a viable solution.

**Objectives:**

1. Translate existing scripts to their `xtask-wasm` equivalents:
    - `examples/image-classification-web/build-for-web.sh`
    - `examples/image-classification-web/run-server.sh`
    - `examples/mnist-inference-web/build-for-web.sh`
    - `examples/mnist-inference-web/run-server.sh`

**About `xtask-wasm`:**

Features of `xtask-wasm` include:

- **Dist**: Ability to generate a distributed package for Wasm.
- **Watch**: Automatically re-run a given command when changes are detected (this leverages `xtask-watch`).
- **DevServer**: Serve the project at a designated IP address.

**Reason for Integration:**

The primary incentive for integrating `xtask-wasm` is its cross-platform capabilities. Ensuring our project's scripts are compatible across different platforms will significantly boost accessibility and usability.

We would appreciate insights, suggestions, or any potential challenges foreseen in this transition.

**Comment by antimora** [comment]
@Luni-4, wanna give a try? =)

**Comment by antimora** [comment]
No, it's not urgent. I just could not complete it in time for the example. I'll be traveling soon so I can't finish it.

**Comment by Luni-4** [comment]
@antimora 

We can rewrite the https://github.com/burn-rs/burn/blob/main/examples/image-classification-web/https_server.py in https://github.com/tokio-rs/axum in order to simplify the `xtask-wasm` transition. Are you willing to do that?

**Comment by Luni-4** [comment]
We can also use https://trunkrs.dev/ instead of `xtask-wasm`. What do you think @antimora?

**Comment by antimora** [comment]
> @antimora 
> 
> 
> 
> We can rewrite the https://github.com/burn-rs/burn/blob/main/examples/image-classification-web/https_server.py in https://github.com/tokio-rs/axum in order to simplify the `xtask-wasm` transition. Are you willing to do that? 

In the end, I think you might end up recreating xtask-wasm. It is better to give a try to use the lib. If we can get it working, the same solution can be applied to other projects (outside burn), including mnist digits.

**Comment by Luni-4** [comment]
> > @antimora
> > We can rewrite the https://github.com/burn-rs/burn/blob/main/examples/image-classification-web/https_server.py in https://github.com/tokio-rs/axum in order to simplify the `xtask-wasm` transition. Are you willing to do that?
> 
> In the end, I think you might end up recreating xtask-wasm. It is better to give a try to use the lib. If we can get it working, the same solution can be applied to other projects (outside burn), including mnist digits.

The `Start` command runs what we have previously defined in `Dist`. In our case, building the web app and running the server are two distinct phases. So I suppose we have to implement a server and then run it, but ok, let me try using the default options and see if we will obtain the same results

**Comment by antimora** [comment]
Related: https://github.com/burn-rs/burn/issues/917

**Comment by antimora** [comment]
No, we need to stick with xtask because it is most portable. We have image classifier that does not use python. And I predict we might want to remove python dependency entirely in the future.

**Comment by nathanielsimard** [comment]
I think having minimal dependency for the build process is very valuable and will contribute to a simpler setup. We use Python in `burn-dataset` to download data from Huggingface, and it often causes problems, but we don't have any other viable alternative for now.

**Comment by antimora** [comment]
@Luni-4 has found this build tool we should look into: https://github.com/trunk-rs/trunk

Here is their vanilla js example: https://github.com/trunk-rs/trunk/tree/main/examples/vanilla

**Comment by Luni-4** [comment]
The documentation and some information are here: https://trunkrs.dev/
